File: extract_pdf.js
const fs = require('fs');
const pdf = require('pdf-parse');

async function extract() {
    for (const file of process.argv.slice(2)) {
        console.log(`\n\n--- Content of ${file} ---`);
        let dataBuffer = fs.readFileSync(file);
        try {
            const data = await pdf(dataBuffer);
            console.log(data.text);
        } catch (e) {
            console.error(`Error reading ${file}:`, e.message);
        }
    }0
}
extract();

File: extract_pdf.py
import sys
from pypdf import PdfReader

def extract_text(pdf_path, out_file):
    out_file.write(f"\n\n--- Content of {pdf_path} ---\n")
    try:
        reader = PdfReader(pdf_path)
        for page in reader.pages:
            text = page.extract_text()
            if text:
                out_file.write(text + "\n")
    except Exception as e:
        out_file.write(f"Error reading {pdf_path}: {e}\n")

if __name__ == "__main__":
    with open("pdf_contents.utf8.txt", "w", encoding="utf-8") as f:
        for path in sys.argv[1:]:
            extract_text(path, f)
